R/as_draws_array.R

```diff
@@ -57,7 +57,11 @@ as_draws_array.draws_df <- function(x, ...) {
   chains <- chain_ids(x)
   out <- vector("list", length(chains))
   for (i in seq_along(out)) {
-    out[[i]] <- x[x$.chain == i, ]
+    if (length(chains) == 1) {
+      out[[i]] <- x
+    } else {
+      out[[i]] <- x[x$.chain == i, ]      
+    }
```

Comment:
The biggest bottleneck now is the as.matrix call below, but that probably cant be avoided.

Reply:
Indeed. We have to transform a list-like object (columns are elements of a list) into a numeric matrix in column major order. I am not sure we can do this faster than base R here.

Reply:
Yeah, I figured.